client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -367,16 +373,38 @@ const PatternAssembler = ( {
 
 	const onSubmit = () => {
 		const design = getDesign();
+		const stylesheet = design?.recipe?.stylesheet ?? '';
+		const themeId = getThemeIdFromStylesheet( stylesheet );
 
-		if ( ! siteSlugOrId ) {
+		if ( ! siteSlugOrId || ! site?.ID || ! themeId ) {
 			return;
 		}
 
-		setPendingAction( () =>
-			applyThemeWithPatterns( siteSlugOrId, design, syncedGlobalStylesUserConfig ).then(
-				( theme: ActiveTheme ) => reduxDispatch( setActiveTheme( site?.ID || -1, theme ) )
-			)
-		);
+		if ( isEnabled( 'pattern-assembler/logged-in-showcase' ) ) {
+			setPendingAction( () =>
+				Promise.resolve()
+					.then( () =>
+						reduxDispatch(
+							activateOrInstallThenActivate( themeId, site?.ID, 'assembler', false, false )
+						)
+					)
+					.then( () =>
+						assembleSite( siteSlugOrId, isSiteJetpack ? themeId : stylesheet, {
+							homeHtml: sections.map( ( pattern ) => pattern.html ).join( '' ),
+							headerHtml: header?.html,
+							footerHtml: footer?.html,
+							globalStyles: syncedGlobalStylesUserConfig,
+							shouldResetContent: isNewSite,
+						} )
+					)
+			);
+		} else {
+			setPendingAction( () =>
+				applyThemeWithPatterns( siteSlugOrId, design, syncedGlobalStylesUserConfig ).then(
```

Comment:
Just noting that `applyThemeWithPatterns` is [used by virtual themes](https://github.com/search?q=repo%3AAutomattic%2Fwp-calypso%20applyThemeWithPatterns&type=code) in general and tailored flows (`link-in-bio` and `free`) without passing the param `sourceSiteId` so the default value (blankcanvas3demo.wordpress.com) is used from this [constant](https://github.com/Automattic/wp-calypso/blob/b63c4cc849f4ea7fe78a7c26877a1ef2a9198cb7/packages/data-stores/src/site/constants.ts#L3).

This means that all the posts from `blankcanvas3demo` site are copied in the user's site because `trimContent` is [hardcoded](https://github.com/Automattic/wp-calypso/blob/b63c4cc849f4ea7fe78a7c26877a1ef2a9198cb7/packages/data-stores/src/site/actions.ts#L562) to be `false` because otherwise only three posts will be copied. 

For sites assembled with the theme `blankcanvas3` we copy posts from `blankcanvas3demo` but for other themes maybe we should use their demo sites. The new endpoint `/site-assembler` handles that we should use it for virtual themes too. cc: @mmtr @taipeicoder